runtime/src/main/java/org/corfudb/runtime/view/stream/BackpointerStreamView.java

```diff
@@ -471,40 +470,27 @@ protected boolean fillReadQueue(final long maxGlobal,
         }
 
         // The maximum address we will fill to.
-        final long maxAddress =
-                Long.min(maxGlobal, context.maxGlobalAddress);
+        final long maxAddress = Long.min(maxGlobal, context.maxGlobalAddress);
 
-        // If we already reached maxAddress ,
-        // we return since there is nothing left to do.
+        // If we already reached maxAddress, we return since there is nothing left to do.
         if (context.getGlobalPointer() >= maxAddress) {
             return false;
         }
 
-        // If everything is available in the resolved
-        // queue, use it
+        // If everything is available in the resolved queue, use it
         if (context.maxResolution >= maxAddress
                 && context.minResolution < context.getGlobalPointer()) {
             return fillFromResolved(maxGlobal, context);
         }
 
-        Long latestTokenValue = null;
+        // Get the stream tail from sequencer.
+        long latestTokenValue = runtime.getSequencerView().query(context.id)
+                .getToken().getSequence();
```

Comment:
In prev version you usually don't go over the wire to get `latestTokenValue` in your version we always ask sequencer for a value. 
Prev version seems better to me. Can you please explain what's the improvement here?

Reply:
If you think about it, that code path in the previous version will never be executed.  maxResolution will always >= greatest value in resolved queue, thus lastTokenValue will always be null and then go to sequencer. If you don't get it, come to me and I will explain.

Reply:
it is not `>=` it is `!=`, correct me if I am wrong, it seems you always execute `latestTokenValue = context.resolvedQueue.ceiling(maxGlobal)` and it equals null only when `ceiling()` return null

Reply:
Oh, ok, got your point
If you are pretty sure that: maxResolution will always >= greatest value in resolved queue
then ok

can we write a unit test for that?

Reply:
`latestTokenValue = context.resolvedQueue.ceiling(maxGlobal)` will always return null at this point, otherwise you already return in the previous code block.

Reply:
If it's not the case, current test cases will catch it. One new test case cannot catch every case.